src/query/boolean_query/mod.rs

```diff
@@ -177,6 +177,40 @@ mod tests {
         }
     }
 
+    #[test]
+    pub fn test_boolean_query_two_excluded() {
+        let (index, text_field) = aux_test_helper();
+
+        let make_term_query = |text: &str| {
+            let term_query = TermQuery::new(
+                Term::from_field_text(text_field, text),
+                IndexRecordOption::Basic,
+            );
+            let query: Box<dyn Query> = Box::new(term_query);
+            query
+        };
+
+        let reader = index.reader().unwrap();
+
+        let matching_docs = |boolean_query: &dyn Query| {
+            reader
+                .searcher()
+                .search(boolean_query, &TEST_COLLECTOR_WITH_SCORE)
+                .unwrap()
+                .docs()
+                .iter()
+                .cloned()
+                .map(|doc| doc.1)
+                .collect::<Vec<DocId>>()
+        };
+        let boolean_query = BooleanQuery::from(vec![
```

Comment:
this test is not testing for this change does it?

Can you add an assert on the score?

Reply:
Testing for the change is somewhat hard because it is testing for doing nothing, and indeed this test does not do that. I did verify that the DoNothing... score combiner is used by adding a println!("...") in its update() method and see that output for the test, but I left that println out of the PR.

I updated the test to compare the score value of a matching doc for an excluding query and a non excluding query.

Reply:
oh yes that makes sense. Thanks!